[BUG] status now shows box as 'down' even though it is up

NAV 5.1:
pping detects a box is down and then up again, but the status still shows 'boxDown' or 'boxShadow'

**To Reproduce**
Steps to reproduce the behavior:
1. pping marks a netbox as down
2. pping marks a netbox as up
3. the status now still has an alert on the box

**Environment (please complete the following information):**
 - chrome 64
 - Debian 10
 - NAV version 5.1.0-2
 - from Debian package
- we're using the exclude_groups and include_groups directives

This might be related: we're seeing a lot of unhandled exceptions in eventengine.log
e.g.:
2020-12-03 06:09:25,479 [ERROR] [nav.eventengine.engine] Unhandled exception while handling boxState end event for odont-do\
-sw5-4.nett (subid=) from pping to eventEngine at 2020-12-03 06:09:25.425425, deleting event
Traceback (most recent call last):
  File "/opt/venvs/nav/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "alerthistmsg_alerthistid_state_msgtype_lan\
guage_key"
DETAIL:  Key (alerthistid, state, msgtype, language)=(293431, e, sms, en) already exists.

--- a/python/nav/eventengine/alerts.py
+++ b/python/nav/eventengine/alerts.py
@@ -166,14 +166,16 @@ class AlertGenerator(dict):
     def _post_alert_messages(self, obj):
         msg_class = obj.messages.model
         for details, text in self._make_messages():
-            msg = msg_class(type=details.msgtype,
-                            language=details.language,
-                            message=text)
-            if hasattr(msg_class, 'alert_queue'):
-                msg.alert_queue = obj
-            elif hasattr(msg_class, 'alert_history'):
-                msg.alert_history = obj
-                msg.state = self.state
+            kwargs = {"type": details.msgtype, "language": details.language}
+            if hasattr(msg_class, "alert_queue"):
+                kwargs["alert_queue"] = obj
+            elif hasattr(msg_class, "alert_history"):
+                kwargs["alert_history"] = obj
+                kwargs["state"] = self.state
+
+            msg, _created = msg_class.objects.get_or_create(
+                **kwargs, defaults={"message": text},
+            )
             msg.save()
 
     def _make_messages(self):

